agents/agents-core/src/commonTest/kotlin/ai/koog/agents/core/dsl/extension/AIAgentNodesHistoryCompressionTest.kt

```diff
@@ -68,6 +69,31 @@ class AIAgentNodesHistoryCompressionTest {
             }
         }
 
+        val myStrategy = simpleStrategy("loop") {
+            while (true) {
+                val response = callLLM()
+                onAssistantMessage(response) {
+                    return@simpleStrategy it.content
+                }
+                val tools = extractTools(response)
+
+                if (latestTokenUsage() > 100500) {
+                    compressHistory()
+                }
+
+                callTools(tools)
+            }
+
+            "Should not be reached"
+        }
```

Comment:
That's really interesting. I was thinking recently that one drawback of the graph DSL was that you have to know in advance the types of the nodes you need. And when write/edit a strategy, you find yourself moving back and forth between your nodes and your edges to adjust types.

This is mainly happening when you use passthrough nodes to try to avoid executing twice the same conditional logic in your edges. This is also happening when you want to do type-safe parallelism.

Would the Koog project be interested in having a second DSL to allow "code" strategies? We could have a `codeStrategy() { }` factory method to handle this case.

I'd be interested in experimenting on this.